@xen-orchestra/rest-api/src/srs/sr.controller.mts

```diff
@@ -79,4 +81,42 @@ export class SrController extends XapiXoController<XoSr> {
 
     return this.sendObjects(Object.values(alarms), req, 'alarms')
   }
+
+  /**
+   * Import an exported VDI
+   * @example id "c4284e12-37c9-7967-b9e8-83ef229c3e03"
+   * @example name_label "VDI_foo_import"
+   * @example name_description "VDI imported by the REST API"
+   * @example raw true
+   */
+  @Example(vdiId)
+  @Post('{id}/vdis')
+  @Tags('vdis')
+  @SuccessResponse(createdResp.status, 'VDI imported')
+  @Response(notFoundResp.status, notFoundResp.description)
+  async srImportVdi(
+    @Request() req: ExRequest & { length?: number },
+    @Path() id: string,
+    @Query() name_label?: string,
+    @Query() name_description?: string,
+    @Query() raw?: boolean
+  ): Promise<{ id: Unbrand<XoVdi>['id'] }> {
+    const xapiSr = this.getXapiObject(id as XoSr['id'])
+    const xapi = xapiSr.$xapi
+
+    if (req.headers['content-length'] !== undefined) {
```

Comment:
```suggestion
    if (this.getHeader('content-length') !== undefined) {
```
To be consistent with existing controller methods, I saw you can use the method getHeader